libkbfs/md_ops.go

```diff
@@ -170,84 +170,85 @@ func (md *MDOpsStandard) processMetadata(ctx context.Context,
 			rmds.SigInfo.VerifyingKey, rmds.untrustedServerTimestamp)
 	}
 	if err != nil {
-		return md.convertVerifyingKeyError(ctx, rmds, err)
+		return ImmutableRootMetadata{}, md.convertVerifyingKeyError(ctx, rmds, handle, err)
 	}
 
 	err = rmds.VerifyRootMetadata(codec, crypto)
 	if err != nil {
-		return err
+		return ImmutableRootMetadata{}, err
+	}
+
+	rmd := RootMetadata{
+		BareRootMetadata: rmds.MD,
+		tlfHandle:        handle,
 	}
 
 	// Try to decrypt using the keys available in this md.  If that
 	// doesn't work, a future MD may contain more keys and will be
 	// tried later.
-	err = decryptMDPrivateData(ctx, md.config, &rmds.MD, &rmds.MD)
+	err = decryptMDPrivateData(
+		ctx, md.config, &rmd, MakeConstRootMetadata(&rmd))
 	if err != nil {
-		return err
+		return ImmutableRootMetadata{}, err
 	}
 
-	return nil
+	mdID, err := md.config.Crypto().MakeMdID(&rmd.BareRootMetadata)
+	if err != nil {
+		return ImmutableRootMetadata{}, err
+	}
+
+	return MakeImmutableRootMetadata(&rmd, mdID), nil
 }
 
 func (md *MDOpsStandard) getForHandle(ctx context.Context, handle *TlfHandle,
-	mStatus MergeStatus) (
-	*RootMetadata, error) {
+	mStatus MergeStatus) (TlfID, ImmutableRootMetadata, error) {
 	mdserv := md.config.MDServer()
 	bh, err := handle.ToBareHandle()
 	if err != nil {
-		return nil, err
+		return TlfID{}, ImmutableRootMetadata{}, err
 	}
 
 	id, rmds, err := mdserv.GetForHandle(ctx, bh, mStatus)
 	if err != nil {
-		return nil, err
+		return TlfID{}, ImmutableRootMetadata{}, err
 	}
 
 	if rmds == nil {
 		if mStatus == Unmerged {
-			// don't automatically create unmerged MDs
-			return nil, nil
-		}
-		var rmd RootMetadata
-		err := updateNewRootMetadata(&rmd, id, bh)
-		if err != nil {
-			return nil, err
+			return TlfID{}, ImmutableRootMetadata{}, nil
```

Comment:
Just wanted to make explicit that the id argument is also empty in the unmerged case. Also added a comment saying that the caller ignores the id argument.